### PR TITLE
VRM WIP Procedural IK Based Animation System

### PR DESCRIPTION
**Pull request by AidanCaruso**
## Summary

A full replacement for the current avatar animation system that uses a unified, procedural IK target solution for all avatar animations, ranging from traditional locomotion to XR.

## References

closes #_insert number here_

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

File: packages/engine/src/avatar/AnimationManager.ts [hunk 1]
```diff
@@ -10,39 +10,13 @@ import { findRootBone, processRootAnimation } from './animation/Util'
...
-  leftHandTarget: Object3D
-  rightFootTarget: Object3D
-  leftFootTarget: Object3D
-}
-
 //Create all IK targets as object 3ds, stored in
 //a named struct and in an object 3d hierarchy
 //the former allows easy accessability while the
 //latter allows for threejs keyframe animation
 export const animationManager = defineState({
```
Comment (HexaField) [inline packages/engine/src/avatar/AnimationManager.ts:17]: Please capitalize

File: packages/engine/src/avatar/animation/TwoBoneIKSolver.ts [hunk 2]
```diff
@@ -36,7 +44,7 @@ export function solveTwoBoneIK(
   targetPosition: Vector3, // world space
   targetRotation: Quaternion, // world space
   rotationOffset: Quaternion | null = null,
-  hint: Object3D | null = null, // todo: in local space, should be in world space, convert to matrix or dual quat
+  hint: Matrix4 | null = null, // todo: in local space, should be in world space, convert to matrix or dual quat
```
Comment (HexaField) [inline packages/engine/src/avatar/animation/TwoBoneIKSolver.ts]: if this is now in world space, please update the comment to reflect this

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/avatar/AnimationManager.ts (hunk 1 above)
Comment (AidanCaruso) [reply packages/engine/src/avatar/AnimationManager.ts:17]: Got it, I still get occasionally tripped up in figuring when to use camelcase or pascal

**Review by speigg — APPROVED** [review]

**Comment by speigg** [comment]
Looks good, please just fix remaining typescript issues and fix the scenes that are pointing to renamed bot avatars. Great job!

**Comment by speigg** [comment]
Closed in favor of https://github.com/EtherealEngine/etherealengine/tree/vrm-mediapipe-update

**Pull request closed without merging**